### PR TITLE
added vue modal component and indicative css

### PR DESCRIPTION
**Pull request by 98dfsj**
# Closes issue: EN-455

## Overview

Adds modal Vue component

## Work carried out

- [x] Added modal component
- [x] Added storybook stories
- [x] Added `open` class with sample css (just turns the modal on/off for now)

## Developer Notes

@hxltrhuxze Some CSS work is needed on here to add the transitions as right now it just shows/hides.

### CODE REVIEW

File: packages/css-framework/src/components/_modal.scss [hunk 1]
```diff
@@ -10,6 +10,14 @@
   display: flex;
   padding: spacing(8);
   position: fixed;
+  opacity:0;
```
Comment (m7kvqbe1) [inline packages/css-framework/src/components/_modal.scss]:
```suggestion
  opacity: 0;
```

**Comment by m7kvqbe1** [comment]
Looks good to me as a basic initial implementation.

One thing to note there is a requirement to have modals which include collections of components like a web form.

Maybe this could be implemented using slots? Allowing us to put anything we like inside a modal.

**Comment by 98dfsj** [comment]
That's already how it works? @m7kvqbe1

**Comment by m7kvqbe1** [comment]
I just realised it doesn't have any tests - is it worth including a couple of simple tests?

**Comment by hxltrhuxze** [comment]
Looks good to me - Do we want the CSS stuff done separately at a later date?

**Comment by 98dfsj** [comment]
@hxltrhuxze It's up to you. You can raise your own PR or I can keep this alive and you can add your bits now if you have time?

**Pull request merged**